src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_GitHubFlow.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    [Category("PostDeployment")]
+    public class ScenarioTests_GitHubFlow : MaestroScenarioTestBase
+    {
+        private TestParameters _parameters;
+        private EndToEndFlowLogic testLogic;
```

Comment:
This field is confusing... This class IS a MaestroScenarioTestBase, but so is EndToEndFlowLogic, so it also CONTAINS a MaestroScenarioTestBase... it's highly unusual to both inherit from a class and have a member that is also of that same base class.  That indicates that something strange is going on in the code sharing department.  One of these two things shouldn't be true, and it means we probably need to rejigger some of the code sharing that's going on so that either EndToEndFlowLogic doesn't inherit from it, to the test fixutres don't.

Code repetition isn't always a bad thing in tests.  Having to read a bunch of types and go through a bunch of base classes to figure out what a test is doing makes it hard to figure out why it fails.

Reply:
It's possible that the test cases don't need to now since they don't actually contain any logic. I'll try to get that cleaned up, I think it's just a result of the refactoring I did to avoid writing the same code 9 times.